reference/reference.go

```diff
@@ -82,9 +82,17 @@ type Spec struct {
 }
 
 var splitRe = regexp.MustCompile(`[:@]`)
+var schemeRe = regexp.MustCompile(`^[a-z0-9\+\-\.]+:.`)
 
 // Parse parses the string into a structured ref.
 func Parse(s string) (Spec, error) {
+	if idx := schemeRe.FindStringIndex(s); idx != nil {
```

Comment:
wondering if checking for `://` would be sufficient  (`if strings.Contains(s, "://")`), as its only for detection, and we're not really interested if the scheme is "valid"

Reply:
Good point. I was initially trying to cover schemes that don't have `//` such as `mailto:` and `news:` but eventually gave up. https://tools.ietf.org/html/rfc1738#page-19.

Reply:
Yes, I think for this case, it's not important to be "fully correct"; we're only interested in "any scheme"; it's an enhancement over the current validation, so it's likely ok if a `mailto:`, `javascript:` or `news:` would accidentally fall through (I expect those to fail further down as well)